src/main/java/com/github/princesslana/eriscasper/rest/guild/CreateGuildRoleRequest.java

```diff
@@ -0,0 +1,30 @@
+package com.github.princesslana.eriscasper.rest.guild;
+
+import java.util.Optional;
+import org.immutables.value.Value;
+
+/**
+ * @see <a
+ *     href="https://discordapp.com/developers/docs/resources/guild#create-guild-role-json-params">
+ *     https://discordapp.com/developers/docs/resources/guild#create-guild-role-json-params</a>
+ */
+@Value.Immutable
+public interface CreateGuildRoleRequest {
+  Optional<String> getName();
+
+  Optional<Integer> getPermissions();
+
+  Optional<Integer> getColor();
+
+  Optional<Boolean> isHoist();
+
+  Optional<Boolean> isMentionable();
+
+  static CreateGuildRoleRequest newRole() {
```

Comment:
Would would we expect this to be used for?

Reply:
Yeah, I mean just making a standard empty new role to work with later, works the same way in the client, just creates a default role to work with and position. If you want I can remove it.

Reply:
Do we think people would create a new role and update it with a bot? Or just create the role with the details they need in the create request?

Reply:
You're right, I'll remove it. Most likely situation is creating with information.